go/datas/database_server.go

```diff
@@ -59,23 +60,32 @@ func (s *RemoteDatabaseServer) Run() {
 	d.Chk.NoError(err)
 	log.Printf("Listening on port %d...\n", s.port)
 
-	router := httprouter.New()
-
-	router.POST(constants.GetRefsPath, s.corsHandle(s.makeHandle(HandleGetRefs)))
-	router.GET(constants.GetBlobPath, s.corsHandle(s.makeHandle(HandleGetBlob)))
-	router.OPTIONS(constants.GetRefsPath, s.corsHandle(noopHandle))
-	router.POST(constants.HasRefsPath, s.corsHandle(s.makeHandle(HandleHasRefs)))
-	router.OPTIONS(constants.HasRefsPath, s.corsHandle(noopHandle))
-	router.GET(constants.RootPath, s.corsHandle(s.makeHandle(HandleRootGet)))
-	router.POST(constants.RootPath, s.corsHandle(s.makeHandle(HandleRootPost)))
-	router.OPTIONS(constants.RootPath, s.corsHandle(noopHandle))
-	router.POST(constants.WriteValuePath, s.corsHandle(s.makeHandle(HandleWriteValue)))
-	router.OPTIONS(constants.WriteValuePath, s.corsHandle(noopHandle))
-	router.GET(constants.BasePath, s.corsHandle(s.makeHandle(HandleBaseGet)))
-
-	router.GET(constants.GraphQLPath, s.corsHandle(s.makeHandle(HandleGraphQL)))
-	router.POST(constants.GraphQLPath, s.corsHandle(s.makeHandle(HandleGraphQL)))
-	router.OPTIONS(constants.GraphQLPath, s.corsHandle(noopHandle))
+	// Note: we implement trailing trailing-slash removal ourselves because
+	// httprouter doesn't handle the redirects properly with CORS.
+	router := httprouter.Router{}
```

Comment:
All false, I mean.

See https://github.com/julienschmidt/httprouter/blob/master/router.go#L170
* We don't want any of the Redirect options (they're all CORS incompatible). 
* HandleMethodNotAllowed is weird
* HandleOPTIONS is irrelevant since we implement our own